From 2b039a235405bfd36b0752a644ee7c342279c0a1 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Mon, 6 Feb 2023 10:14:32 +0800
Subject: [PATCH 38/45] Revert "amdgpu/pm: prevent array underflow in
 vega20_odn_edit_dpm_table()"

This reverts commit 85273b4a7076ed5328c8ace02234e4e7e10972d5.
---
 drivers/gpu/drm/amd/pm/powerplay/hwmgr/vega20_hwmgr.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/pm/powerplay/hwmgr/vega20_hwmgr.c b/drivers/gpu/drm/amd/pm/powerplay/hwmgr/vega20_hwmgr.c
index 57a354a03e8ae..60cde0c528257 100644
--- a/drivers/gpu/drm/amd/pm/powerplay/hwmgr/vega20_hwmgr.c
+++ b/drivers/gpu/drm/amd/pm/powerplay/hwmgr/vega20_hwmgr.c
@@ -2962,8 +2962,7 @@ static int vega20_odn_edit_dpm_table(struct pp_hwmgr *hwmgr,
 			data->od8_settings.od8_settings_array;
 	OverDriveTable_t *od_table =
 			&(data->smc_state_table.overdrive_table);
-	int32_t input_clk, input_vol, i;
-	uint32_t input_index;
+	int32_t input_index, input_clk, input_vol, i;
 	int od8_id;
 	int ret;
 
-- 
2.25.1

